op-e2e/system_test.go

```diff
@@ -1371,3 +1371,43 @@ func latestBlock(t *testing.T, client *ethclient.Client) uint64 {
 	require.Nil(t, err, "Error getting latest block")
 	return blockAfter
 }
+
+// TestPendingBlockIsLatest tests that we serve the latest block as pending block
+func TestPendingBlockIsLatest(t *testing.T) {
+	InitParallel(t)
+
+	cfg := DefaultSystemConfig(t)
+	sys, err := cfg.Start()
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+	l2Seq := sys.Clients["sequencer"]
+
+	t.Run("block", func(t *testing.T) {
+		for i := 0; i < 10; i++ {
+			pending, err := l2Seq.BlockByNumber(context.Background(), big.NewInt(-1))
```

Comment:
We should use `rpc.PendingBlockNumber` over `-1`, based on https://github.com/ethereum/go-ethereum/pull/27219 it seems like the numbers do not have a guarantee of stability

Reply:
I tried that, but the server-side magic numbers do not match the client-side magic numbers.

Reply:
Do the numbers match after that PR switches them? This looks like we will have to change this when upstream changes.

Reply:
Probably worth declaring it as a constant/var locally just for clarity of purpose if we can't use an existing constant.

The only real risk is if -1 winds up meaning `latest` at some point in the future but I suspect that would cause a lot of grief for anyone using the `ethclient` module given it does indeed use hard coded `-1` when doing its translation and then it's actually `pending` that gets sent in the RPC request.

Reply:
Added TODOs with reference to ticket of next geth update, which would include this upstream change.